src/document_basic.c

```diff
@@ -417,54 +409,53 @@ void Document_LoadPairwiseArgs(Document *d, RedisModuleString **args, size_t nar
-    dst->name = name;
+    dst->docFieldName = NewHiddenName(name, strlen(name), false);
     dst->text = args[ii + 1];
     dst->unionType = FLD_VAR_T_RMS;
   }
 }
 
-void Document_Clear(Document *d) {
-  if (d->flags & (DOCUMENT_F_OWNSTRINGS | DOCUMENT_F_OWNREFS)) {
-    for (size_t ii = 0; ii < d->numFields; ++ii) {
-      DocumentField *field = &d->fields[ii];
-      if (d->flags & DOCUMENT_F_OWNSTRINGS) {
-        rm_free((void *)field->name);
-        if (field->path && (field->path != field->name)) {
-          rm_free((void *)field->path);
-        }
-      }
-      switch (field->unionType) {
-        case FLD_VAR_T_RMS:
-          RedisModule_FreeString(RSDummyContext, field->text);
-          break;
-        case FLD_VAR_T_CSTR:
-          rm_free(field->strval);
-          break;
-        case FLD_VAR_T_ARRAY:
-        // TODO: GEOMETRY Handle multi-value geometry fields
-          if (field->indexAs & (INDEXFLD_T_FULLTEXT | INDEXFLD_T_TAG | INDEXFLD_T_GEO)) {
-            for (int i = 0; i < field->arrayLen; ++i) {
-              rm_free(field->multiVal[i]);
-            }
-            rm_free(field->multiVal);
-            field->arrayLen = 0;
-          } else if (field->indexAs & INDEXFLD_T_NUMERIC) {
-            array_free(field->arrNumval);
-          }
-          if (field->multisv) {
-            RSValue_Free(field->multisv);
+void ClearOwnedField(DocumentField *field) {
+  switch (field->unionType) {
+    case FLD_VAR_T_RMS:
+      RedisModule_FreeString(RSDummyContext, field->text);
+    break;
+    case FLD_VAR_T_CSTR:
+      rm_free(field->strval);
+    break;
+    case FLD_VAR_T_ARRAY:
+      // TODO: GEOMETRY Handle multi-value geometry fields
+        if (field->indexAs & (INDEXFLD_T_FULLTEXT | INDEXFLD_T_TAG | INDEXFLD_T_GEO)) {
+          for (int i = 0; i < field->arrayLen; ++i) {
+            rm_free(field->multiVal[i]);
           }
-          break;
-        case FLD_VAR_T_BLOB_ARRAY:
-          rm_free(field->blobArr);
-          field->blobArrLen = 0;
-          break;
-        case FLD_VAR_T_GEO:
-        case FLD_VAR_T_NUM:
-        case FLD_VAR_T_GEOMETRY:
-        case FLD_VAR_T_NULL:
-          break;
-      }
+          rm_free(field->multiVal);
+          field->arrayLen = 0;
+        } else if (field->indexAs & INDEXFLD_T_NUMERIC) {
+          array_free(field->arrNumval);
+        }
+    if (field->multisv) {
+      RSValue_Free(field->multisv);
+    }
+    break;
+    case FLD_VAR_T_BLOB_ARRAY:
+      rm_free(field->blobArr);
+    field->blobArrLen = 0;
+    break;
+    case FLD_VAR_T_GEO:
+    case FLD_VAR_T_NUM:
+    case FLD_VAR_T_GEOMETRY:
+    case FLD_VAR_T_NULL:
+      break;
+  }
+}
+
+void Document_Clear(Document *d) {
+  for (size_t ii = 0; ii < d->numFields; ++ii) {
+    DocumentField *field = &d->fields[ii];
+    HiddenName_Free(field->docFieldName, d->flags & DOCUMENT_F_OWNSTRINGS);
+    if (d->flags & (DOCUMENT_F_OWNSTRINGS | DOCUMENT_F_OWNREFS)) {
```

Comment:
Notice this condition was in the beginning before. Do we now always have a docFieldName?

Reply:
Yes, the flags are now more related to whether the field is owned by us.
I can maybe think about removing `DOCUMENT_F_OWNSTRINGS`.
WDYT?

Reply:
OK.. A bit weird but not critical.